packages/gatsby-remark-images/src/index.js

```diff
@@ -282,10 +282,20 @@ module.exports = (
     const imageCaption =
       options.showCaptions && getImageCaption(node, overWrites)
 
+    let removeBgImage = false
+    if (options.disableBgImageOnAlpha) {
+      const imageStats = await stats({ file: imageNode, reporter })
+      if (imageStats && imageStats.isTransparent) removeBgImage = true
+    }
+
+    const bgImage = !removeBgImage
+      ? ` background-image: url('${placeholderImageData}');`
+      : ``
+
     let rawHTML = `
   <span
     class="${imageBackgroundClass}"
-    style="padding-bottom: ${ratio}; position: relative; bottom: 0; left: 0; background-image: url('${placeholderImageData}'); background-size: cover; display: block;"
+    style="padding-bottom: ${ratio}; position: relative; bottom: 0; left: 0;${bgImage} background-size: cover; display: block;"
```

Comment:
we can move `background-size: cover;` also inside the bgImage variable as it won't have any effect when there is no background-image set.

Reply:
i moved `background-size: cover;` inside the `bgImage` variable